utils/open_close_files_benchmark.cc

```diff
@@ -0,0 +1,133 @@
+/*
+   Copyright 2023 Leil Storage OÜ
+
+   This file is part of SaunaFS.
+
+   SaunaFS is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, version 3.
+
+   SaunaFS is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with SaunaFS. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <algorithm>
+#include <cstring>
+#include <fcntl.h>
+#include <filesystem>
+#include <iostream>
+#include <unistd.h>
+#include <vector>
+
+void show_help_message_and_exit(char *progname, int status) {
+	std::cerr << "Usage:\n"
+	             "    "
+	          << progname
+	          << " <TESTING_PATH> <NUMBER_OF_FILES> [-r]\n"
+	             "    "
+	          << "Creates a folder 'open_close_files_benchmark_<TIMESTAMP>',\n"
+	             "    "
+	          << "then creates NUMBER_OF_FILES empty files while keeping all\n"
+	             "    "
+	          << "of them open and finally closes all of them.\n"
+	             "    "
+	          << "Option -r is to reverse the order of closing the files.\n\n"
+	             "    "
+	          << "Note: NUMBER_OF_FILES must be positive.\n"
+	          << std::endl;
+	exit(status);
+}
+
+int main(int argc, char **argv) {
+	if (argc < 3 || argc > 4) {
+		show_help_message_and_exit(argv[0], 1);
+	}
+
+	std::string testing_path(argv[1]);
+	int number_of_files = atoi(argv[2]);
```

Comment:
At some point we considered to change the snake casing to semi camel casing for new code. Please consider it.

Reply:
Applied.